### ISSUE TITLE
Latest WineD3D requires new functions

### ISSUE THREAD

**Issue by Svyatpro** [MyTDT-Mysoft/DllCompat#20]
gdi3x.dll
D3DKMTCreateDevice
D3DKMTDestroyDevice
D3DKMTOpenAdapterFromGdiDisplayName
D3DKMTSetVidPnSourceOwner

ntdlx:
strnlen

Latest Wine repo has those functions in gdi32 and in ntdll

**Comment by pachuco** [comment]
Added strnlen.
I am going to assume these functions are for fdossena wined3d.
If so, then you can run wined3d up to v4.18 for now.

**Comment by Svyatpro** [comment]
I can run latest WineD3D on XP for now. And I compile WineD3D far batter than fdossena's build I even have working DirectWrite for XP. All KMT* stuff can be wrapped like One-Core-API did

**Comment by pachuco** [comment]
Is it something that is done in userspace, for the rest of these D3DKMT functions?

**Comment by Svyatpro** [comment]
Yes, and some necessary functions can be just forwarded to DX SDK's d3d11ref.dll =)

For example,
D3DKMTCreateDevice
D3DKMTOpenAdapterFromGdiDisplayName

these functions are vital for WineD3D but they easily can be forwarded to d3d11ref.dll from DX SDK and WineD3D works again without problems

**Comment by pachuco** [comment]
Does this reference device have source in the sdk?

**Comment by pachuco** [comment]
Hmmm, this reference dll depends on CancelIoEx. Never got around to that one.

**Comment by Svyatpro** [comment]
> 

no, there's no source code but we can use it as a "guest dll"

You can search and download older version of d3d11ref.dll it has less NT6 only dependencies
